torch_xla/csrc/convolution.cpp

```diff
@@ -42,41 +45,90 @@ tensorflow::ConvOpAttrs MakeConvOpAttrs(
   return conv_op_attrs;
 }
 
-std::vector<xla::int64> FilterTransposePermutation() { return {2, 3, 1, 0}; }
+// Transpose filter shape to have [channel, batch] as last two dimensions.
+// 4D case: (N, C, H, W) -> (H, W, C, N)
+const std::vector<xla::int64>& FilterTransposePermutation(const xla::int64 k) {
+  XLA_CHECK(k == 4 || k == 5);
```

Comment:
When doing bare `XLA_CHECK()` the value of `k` won't be shown, so maybe emitting it can be of some help.
Even a simple `<< k`.